File: algorithms/code/leetcode/lc009_palindrome_number/lc009_palindrome_number.rs
// https://leetcode.com/problems/palindrome-number/

pub struct Solution {}

impl Solution {
    pub fn is_palindrome(mut x: i32) -> bool {
        if x < 0 {
            return false;
        }
        let xa: i32 = x;
        let mut xb: i32 = 0;
        while x > 0 {
            xb = xb * 10 + x % 10;
            x /= 10;
        }
        xa == xb
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_213() {
        assert!(!Solution::is_palindrome(213));
    }

    #[test]
    fn test_456() {
        assert!(!Solution::is_palindrome(456));
    }

    #[test]
    fn test_454() {
        assert!(Solution::is_palindrome(454));
    }

    #[test]
    fn test_99() {
        assert!(Solution::is_palindrome(99));
    }

    #[test]
    fn test_1() {
        assert!(Solution::is_palindrome(1));
    }

    #[test]
    fn test_10() {
        assert!(!Solution::is_palindrome(10));
    }

    #[test]
    fn test_minus1() {
        assert!(!Solution::is_palindrome(-1));
    }

    #[test]
    fn test_0() {
        assert!(Solution::is_palindrome(0));
    }
}

File: algorithms/code/leetcode/lc009_palindrome_number/lc009_palindrome_number.py
# https://leetcode.com/problems/palindrome-number/

import unittest


class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        xa = x
        xb = 0
        while x > 0:
            xb = xb * 10 + x % 10
            x //= 10
        return xa == xb


class TestCode(unittest.TestCase):
    def test_213(self) -> None:
        assert not Solution().isPalindrome(213)

    def test_456(self) -> None:
        assert not Solution().isPalindrome(456)

    def test_454(self) -> None:
        assert Solution().isPalindrome(454)

    def test_99(self) -> None:
        assert Solution().isPalindrome(99)

    def test_1(self) -> None:
        assert Solution().isPalindrome(1)

    def test_10(self) -> None:
        assert not Solution().isPalindrome(10)

    def test_minus1(self) -> None:
        assert not Solution().isPalindrome(-1)

    def test_0(self) -> None:
        assert Solution().isPalindrome(0)
